checks/raw/security_policy.go

```diff
@@ -112,3 +137,87 @@ func isSecurityPolicyFilename(name string) bool {
 		strings.EqualFold(name, "doc/security.rst") ||
 		strings.EqualFold(name, "docs/security.rst")
 }
+
+var checkSecurityPolicyFileContent fileparser.DoWhileTrueOnFileContent = func(path string, content []byte,
+	args ...interface{},
+) (bool, error) {
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires exactly one argument: %w", errInvalidArgLength)
+	}
+	pfiles, ok := args[0].(*[]checker.File)
+	if !ok {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires argument of type *[]checker.File: %w", errInvalidArgType)
+	}
+
+	if len(content) == 0 {
+		// perhaps there are more policy files somewhere else,
+		// keep looking (true)
+		return true, nil
+	}
+
+	// TODO: there is an assertion here that if content has length
+	//       then (*pfiles)[0] is not nil (can that be checked)
+	//       (so is this test necessary?)
+	if (*pfiles) != nil {
```

Comment:
return an error if it's nil, because this means a bug in our code.

Reply:
Will do!

Reply:
in revising `SecurityPolicyData` to include `SecurityPolicyInformationType`/`SecurityPolicyInformation` I went the route to note for this policy check there is only on `File` (as opposed to `[]File`) - as such I am now checking for empty `File` (i.e., `(File{})`) - but I do have to check for `nil` on `(*pinfo)` for the new policy information.

So .. the intent is done.